src/main.rs

```diff
@@ -1,3 +1,436 @@
+    let line_text = &captures[4];
+    let method_name_match = regex!(r#"^[a-z_]+"#)
+        .captures(&line_text[column..])
+        .unwrap_or_else(|| {
+            panic!("couldn't find invoked method name at column position in line: '{match_text}'")
+        });
+    let mut method_name = &method_name_match[0];
+    if should_strip_raw_suffix {
+        assert!(method_name.ends_with("_raw"));
+        method_name = &method_name[..method_name.len() - 4];
+    }
+    let target_method_definition = target_method_definitions_by_name
+        .get(method_name)
+        .cloned()
+        // .unwrap_or_else(|| panic!("found method name wasn't known: '{method_name}'"));
+        .unwrap_or_else(|| panic!("method_name: {method_name:?}, line_text: {line_text:?}, match_text: {match_text:?}"));
+    MethodCallLocation {
+        location: Location {
+            line,
+            column,
+            file_path,
+        },
+        method_definition: target_method_definition,
+    }
+}
+
+fn parse_target_method_invocations(
+    matches_text: &str,
+    target_method_definitions_by_name: &HashMap<String, MethodDefinitionLocationAndName>,
+    should_strip_raw_suffix: bool,
+) -> Vec<MethodCallLocation> {
+    matches_text
+        .split('\n')
+        .filter(|line| !line.is_empty())
+        .map(|line| {
+            parse_target_method_invocation(
+                line,
+                target_method_definitions_by_name,
+                should_strip_raw_suffix,
+            )
+        })
+        .collect()
+}
+
+fn parse_wrap_invocation(match_text: &str) -> Location {
+    let captures = regex!(r#"^([^:]+):(\d+):(\d+):"#)
+        .captures(match_text)
+        .unwrap_or_else(|| panic!("wrap invocation regex didn't match line: '{match_text}'"));
+    let file_path = captures[1].to_owned();
+    let line: usize = captures[2].parse::<usize>().unwrap() - 1;
+    let column: usize = captures[3].parse::<usize>().unwrap() - 1;
+    Location {
+        line,
+        column,
+        file_path,
+    }
+}
+
+fn parse_wrap_invocations(matches_text: &str) -> Vec<Location> {
+    matches_text
+        .split('\n')
+        .filter(|line| !line.is_empty())
+        .map(parse_wrap_invocation)
+        .collect()
+}
+
+fn remove_wrap_invocation(wrap_invocation: &Location) {
+    remove_range_in_file(
+        &wrap_invocation.file_path,
+        wrap_invocation.line,
+        wrap_invocation.column - 1,
+        wrap_invocation.line,
+        wrap_invocation.column + 6,
+    );
+}
+
+fn remove_wrap_invocations(wrap_invocations: &[Location]) {
+    for wrap_invocation in wrap_invocations {
+        remove_wrap_invocation(wrap_invocation);
+    }
+}
+
 fn main() {
-    println!("Hello, world!");
+    env::set_current_dir("/Users/jrosse/prj/tsc-rust/typescript_rust").unwrap();
+
+    let mut target_method_definitions = get_target_method_definitions();
+    target_method_definitions.sort();
+    target_method_definitions.reverse();
```

Comment:
By being a bit specific about the order in which I was applying this particular set of text edits I think I was able to avoid any nastiness around eg making multiple edits to a file that would then "throw off" previously queried source-code location information etc